client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +139,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
-            Preconditions.checkState(started.compareAndSet(false, true));
+            Preconditions.checkState(!started.get());
```

Comment:
As a general rule, I always encourage adding a meaningful message to any exception being thrown. That message will be added to the log and make debugging easieer.